alchemy/alchemy.py

```diff
@@ -980,31 +1054,87 @@ def _alchemicallyModifyNonbondedForce(self, system, reference_force):
             # Add parameters to custom force handling interactions between alchemically-modified atoms and rest of system.
             sterics_custom_nonbonded_force.addParticle([sigma, epsilon])
             electrostatics_custom_nonbonded_force.addParticle([charge, sigma])
-            # Turn off Lennard-Jones contribution from alchemically-modified particles.
-            if particle_index in alchemical_atom_indices:
-                nonbonded_force.setParticleParameters(particle_index, abs(0*charge), sigma, abs(0*epsilon))
 
         # Move NonbondedForce exception terms for alchemically-modified particles to CustomNonbondedForce/CustomBondForce.
+        exception_pairs = set()
         for exception_index in range(nonbonded_force.getNumExceptions()):
             # Retrieve parameters.
             [iatom, jatom, chargeprod, sigma, epsilon] = nonbonded_force.getExceptionParameters(exception_index)
+            # Record exception/exclusion pairs.
+            exception_pairs.add( (iatom, jatom) )
+            exception_pairs.add( (jatom, iatom) )
             # Exclude this atom pair in CustomNonbondedForce.
             sterics_custom_nonbonded_force.addExclusion(iatom, jatom)
             electrostatics_custom_nonbonded_force.addExclusion(iatom, jatom)
-            # Move exceptions involving alchemically-modified atoms to CustomBondForce.
-            if self.annihilate_sterics and (iatom in alchemical_atom_indices) and (jatom in alchemical_atom_indices):
-                # Add special CustomBondForce term to handle alchemically-modified Lennard-Jones exception.
-                custom_bond_force.addBond(iatom, jatom, [chargeprod, sigma, epsilon])
-                # Zero terms in NonbondedForce.
-                nonbonded_force.setExceptionParameters(exception_index, iatom, jatom, abs(0*chargeprod), sigma, abs(0*epsilon))
+            # Move exceptions involving the annihilation of alchemically-modified atoms to CustomBondForce.
+            # For decoupled sterics or electrostatics, exceptions are retained.
+            if self.annihilate_sterics and ((iatom in alchemical_atom_indices) or (jatom in alchemical_atom_indices)):
```

Comment:
Maybe define `alchemical_atomset` before (now it's defined in line 1083) and check inclusion in set instead of a list?